src/primaite/agents/aegis/data.py

```diff
@@ -22,21 +25,24 @@ def __getitem__(self, idx):
             "question": self.questions[idx],
             "gt_answer": self.gt_answers[idx],
             "gt_hidden_states": self.gt_hidden_states[idx],
+            "gt_logprobs": self.gt_logprobs[idx],
+            "gt_tokens": self.gt_tokens[idx],
         }
 
     def _get_target_hidden_states(self, gt_answers: List[str]):
-        hidden_states = torch.empty(0)
-        for gt_answer in gt_answers:
-            inputs = self.llm.get_input_embeddings(prompts=[gt_answer], grad=False)
-            hidden_states = torch.cat(
-                [
-                    hidden_states,
-                    self.llm.get_output_embeddings(inputs_embeds=inputs["inputs_embeds"], grad=False).to("cpu"),
-                ],
-                dim=0,
-            )
+        inputs = self.llm.get_input_embeddings(prompts=gt_answers, grad=False)
+        hidden_states = self.llm.get_output_embeddings(inputs_embeds=inputs["inputs_embeds"], grad=False).to("cpu")
         return hidden_states
 
+    def _get_target_logprobs_and_tokens(self, gt_answers: List[str]):
+        # Construct mock logprobs for gt repsonse
+        gt_tokens = self.llm.tokenizer.batch_encode_plus(gt_answers, return_tensors="pt", padding=True)["input_ids"]
+        batch_size = gt_tokens.shape[0]
+        gt_probs = torch.zeros((batch_size, gt_tokens.shape[1], len(self.llm.tokenizer)), dtype=torch.float32)
+        for token_idx in range(gt_tokens.shape[1]):
```

Comment:
This loop can be done in one operation using like a torch scatter, but its probably not that performance detrimental, im just being picky, and torch scatter is kinda complicated lol